### ISSUE TITLE
login()  always return true regardless of legacy

### ISSUE THREAD

**Issue by matracine** [EvilFreelancer/routeros-api-php#3]
Hello,
first thanks for this great job!

I've got a lot of routers with versions pre 6.43 and post 6.43.

When connecting to ROS pre 6.43 and setting legacy: false (wich is wrong, but I want to determine the router version), login() method returns true.

Here is the code :
```php
        // Return true if we have only one line from server and this line is !done
        return isset($response[0]) && $response[0] === '!done';
``` 

I think the point is that you don't test if there is only one line in the response.
```php
        // Return true if we have only one line from server and this line is !done
        return (1 === count($response)) &&  isset($response[0]) && ($response[0] === '!done');
``` 

Works for me...

**Comment by matracine** [comment]
Here is a more complete proposition :

```php
        // if :
        //  - we have more than one response
        //  - response is '!done'
        // => problem with legacy version, swap it and retry
        // Only tested with ROS pre 6.43, will test with post 6.43 => this could make legacy parameter obsolete ?
        if (count($response)>1 && $response[0] === '!done' && !$this->config('legacy'))
        {
            $this->_config->set('legacy', true);
            return $this->login();
        }
        // Return true if we have only one line from server and this line is !done
        return (1 === count($response)) &&  isset($response[0]) && ($response[0] === '!done');

```

**Comment by EvilFreelancer** [comment]
@matracine hello! And thank you for this good solution. Maybe you can create Merge Request with your fixes?

**Comment by EvilFreelancer** [comment]
Nice, can you add these few lines to unit tests of Client class?

**Comment by matracine** [comment]
Will try, didn't looked your tests, sorry.

**Comment by EvilFreelancer** [comment]
You can use my [other project](https://github.com/EvilFreelancer/docker-routeros) with RouterOS in Docker container for running unit testing on your computer. You just need to have expect, Docker, Docker Compose, next clone the repo and exec `docker-compose up -d`, then you need preconfigure virtual routers via [preconf.tcl](https://github.com/EvilFreelancer/routeros-api-php/blob/master/preconf.tcl) script from root of routeros-api-php:
```
./preconf.tcl 12223
./preconf.tcl 22223
```
And after this you can run tests:
```
./vendor/bin/phpunit
```

**Comment by matracine** [comment]
Great, I was looking how to do that, thanks. On the way....

**Comment by matracine** [comment]
Almost ready, but I'm not super "pull request aware" ;)
How do I modify my pull request ?
Can you review my branch and tell me if this is OK for you (added test__constructLegacy2() in ClientTest.php)

**Comment by EvilFreelancer** [comment]
Small update by me, I've added endless loop detection logic.
Thanks for your help @matracine! Now $this library can automaticaly detect legacy firmware and this is awesome.

**Comment by matracine** [comment]
Happy if I helped, this is realy a great lib.
Nothing more to do for me ?
Thanks.

**Comment by EvilFreelancer** [comment]
If you want you may help me with other issue #2 it's about queries in loop, sometimes library returned empty array. The second issue is achieving 100% code coverage of code (details is [here](https://www.reddit.com/r/mikrotik/comments/awtx5x/need_help_with_unit_testing_of_api_client_library/)).

By the way, look at [new syntax](https://github.com/EvilFreelancer/routeros-api-php/blob/master/examples/vlans_bridge_v3.php), you can simplify your code little bit.

In any way thank you for help :)

**Comment by EvilFreelancer** [comment]
Oh, I forgot to say, your edits are available in the 0.8.1 version of the library.

**Comment by matracine** [comment]
Hello,
Thanks, saw it for 0.8.1, I already updated my projects :)
I will have a look for #2 and code coverage.

Regards,
